lib/mutant/matcher/method.rb

```diff
@@ -1,133 +1,154 @@
 module Mutant
   class Matcher
-    # Matcher for subjects that are a specific method
+    # Abstract base class for method matchers
     class Method < self
-      include Adamantium::Flat, Concord::Public.new(:env, :scope, :target_method)
-      include AST::NodePredicates
+      include AbstractType,
+              Adamantium::Flat,
+              Concord::Public.new(:scope, :target_method, :evaluator)
 
       # Methods within rbx kernel directory are precompiled and their source
       # cannot be accessed via reading source location. Same for methods created by eval.
       BLACKLIST = %r{\A(kernel/|\(eval\)\z)}.freeze
 
-      # Enumerate matches
+      SOURCE_LOCATION_WARNING =
+        '%s does not have valid source location, unable to emit subject'.freeze
+
+      CLOSURE_WARNING =
+         '%s is dynamically defined in a closure, unable to emit subject'.freeze
```

Comment:
Interesting.  I think you should add an example like the code you specified to the warning.  It will likely end up as a question on github issues otherwise.